### PR TITLE
Issues/345 update structure

### PR DESCRIPTION
**Pull request by pboling**
# No functional runtime changes

Addresses #345 

- [x] Update to current RSpec syntax `RSpec.describe` at outset
- [x] Add `rubocop-rspec` for a suite of RSpec related rules
- [x] Readme updates
  - Update Copyright
  - Add development tips
  - Add contributing tips
  - Add Code of Conduct link
- [x] Update gemspec to define all development dependencies that are across all supported Rubies
- [x] Document the namespace of the library
- [x] Run specs in documentation format and auto-require `spec_helper` which now requires `helper`
  - remove `require 'helper'` lines, no longer needed.
- [x] Enhance `.gitignore` & document it
- [x] Add `CODE_OF_CONDUCT.md`
- [x] Add bin files: `bin/console` and `bin/setup`
- [x] Add new copyright line, current year, to `LICENSE.md`
- [x] Add more supported Rubies to build
  - `jruby-9.0`
  - `jruby-1.7`
  - `ruby-2.5`
- [x] Split out some Cops that we want to progressively adopt, but are currently disabled in `.rubocop_todo.yml`
- [x] Add `wwtd` gem to allow for local simulation of the travis build.

### CODE REVIEW

**Comment by xtagon** [comment]
Alright, the changes look good to me. I especially like the Code of Conduct added. 👍

**Comment by xtagon** [comment]
Any idea why the build has a rake error?

**Comment by pboling** [comment]
@xtagon I need to do some additional gem version constraining by Ruby version.  I'll do that tonight.

File: oauth2.gemspec [hunk 1]
```diff
@@ -10,17 +10,33 @@ Gem::Specification.new do |spec|
...
   spec.required_ruby_version = '>= 1.9.0'
   spec.required_rubygems_version = '>= 1.3.5'
   spec.summary       = 'A Ruby wrapper for the OAuth 2.0 protocol.'
   spec.version       = OAuth2::Version
+
+  spec.require_paths = %w[lib]
+  spec.bindir        = 'exe'
+  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
+    f.match(%r{^(test|spec|features)/})
```
Comment (josephpage) [inline oauth2.gemspec]: The `bin` directory is probably for dev only and should be excluded too.

File: .travis.yml [hunk 2]
```diff
@@ -11,21 +12,40 @@ env:
...
   allow_failures:
     - rvm: jruby-head
     - rvm: ruby-head
   fast_finish: true
+  include:
+#    - rvm: 1.8.7-p371
+#      gemfile: gemfiles/ruby_1.8.gemfile
+    - rvm: jruby-1.7.27
+      gemfile: gemfiles/jruby_1.7.gemfile
+#    - rvm: 1.9.3-p551
```
Comment (josephpage) [inline .travis.yml]: Do you think we should not try to support 1.9 & 2.0 ?

Comment (pboling) [reply .travis.yml]: No, they are only commented out for my local testing.  I will turn them back on once I get things stable.

**Comment by pboling** [comment]
@xtagon This is done!  Mostly the same as when you reviewed.  Removed and ignored the gemfile.locks.  Added rubocop-rspec for more rules.  Turned back on the Rubies I had disabled.

**Comment by xtagon** [comment]
Thanks @pboling

**Pull request merged**